src/src/diag.c

```diff
@@ -65,6 +65,10 @@
 #include "sl_rail_util_ant_div.h"
 #endif
 
+#ifdef SL_CATALOG_OT_CRASH_HANDLER_PRESENT
+#include PLATFORM_HEADER
```

Comment:
Why are we including `PLATFORM_HEADER` here, and what is it used for?
Can we just include the header name directly?

Reply:
`PLATFORM_HEADER` refers to [platform/service/legacy_hal/inc/platform-header.h](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/platform/service/legacy_hal/inc/platform-header.h)

It's included mainly for [these lines](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/platform/service/legacy_hal/inc/platform-header.h#L570-L579). It redefines `assert()` so that the crash handler's `halInternalAssertFailed()` is used instead of standard `assert`, allowing us to save some info about the assert to the crash dump

```c
void halInternalAssertFailed(const char * filename, int linenumber);
#define ASSERT_HANDLER halInternalAssertFailed
...

#undef assert
#define assert(condition)  \
  do { if (!(condition)) { \
         ASSERT_HANDLER(__SOURCEFILE__, __LINE__); } } while (0)
```

Reply:
I'm open to including [legacy_hal/inc/platform-header.h](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/platform/service/legacy_hal/inc/platform-header.h) directly.